catch/external/Catch2/cmake/Catch2/Catch.cmake

```diff
@@ -118,8 +118,10 @@ same as the Catch name; see also ``TEST_PREFIX`` and ``TEST_SUFFIX``.
 
 #]=======================================================================]
 
+
 #------------------------------------------------------------------------------
-function(catch_discover_tests TARGET)
+# TARGET_LIST TEST_SET
+function(catch_discover_tests_compile_time_detection TARGET TEST_SET)
```

Comment:
Can we not change this func name?

This is a standard catch2 macro and will complicate when we decide to update catch2 for HIP tests